### ISSUE TITLE
Can't recover original filenames after predict_on_dataset

### ISSUE THREAD

**Issue by fitzkmic** [ElementAI/baal#29]
**Describe the bug**
If I run the example "Use BaaL in production (Classification)" and I use `top_uncertainty = heuristic(predictions)[:10]` I get indices from the unlabeled pool dataset back. To recover the original filenames from `train_dataset` I need indices on the whole `active_learning_ds` otherwise already labeled data is not taken into consideration correctly.

**To Reproduce**
Try out the example and print `heuristic(prediction)`

**Expected behavior**
`heuristic(predictions)` should give back indices that refer to the `active_learning_ds` not only the `pool`.

**Vesion (please complete the following information):**
 - OS: Ubuntu 16.04
 - Python: 3.7
 - BaaL version: 1.1.1

**Comment by parmidaatg** [comment]
Thank you for bringing up this issue @fitzkmic. To map the pool indices to the dataset indices, we already have [`_pool_to_oracle_index`](https://github.com/ElementAI/baal/blob/master/src/baal/active/dataset.py#L127) in `ActiveLearningDataset` class. We already use this mapping in the `label` method in same class, however, this is a `private` method and reading your issue we realized that it might be a good idea to make it public. For the moment you can simply call the method on the list of indices chosen by heuristics as below: 
`indices = heuristic(predictions)
 indices = trainset._pool_to_oracle_index(indices)
`
Heuristics do not deal with the dataset, all the dataset related matters are handled in `ActiveLearningDataset`.

**Comment by fitzkmic** [comment]
works, thank you!
